lib/httplog/http_log.rb

```diff
@@ -52,10 +53,15 @@ def url_approved?(url)
       !config.url_whitelist_pattern || url.to_s.match(config.url_whitelist_pattern)
     end
 
+    def masked_body_url?(url)
+      config.filter_parameters.any? &&
+        (config.url_masked_body_pattern && url.to_s.match(config.url_masked_body_pattern) || config.mask_json)
```

Comment:
See above - the final condition wouldn't be necessary if `url_masked_body_pattern == /.*/`